### DISCUSSION TITLE
HLS install directories

### DISCUSSION THREAD

**Discussion by cheeze2000** [williamboman/nvim-lsp-installer#842] [Q&A] [answered]
When I installed hls from `:LspInstallInfo`, the install_dir is in `~/.local/share/nvim/lsp_servers/haskell`. Then, I deleted the `lsp_servers` folder and installed `hls` again but using the [debug init file](https://github.com/williamboman/nvim-lsp-installer/blob/main/tests/minimal_debug_init.lua). It now installs it in `~/.local/share/nvim/lsp_servers/hls`. Is there any reason why this happens? I don't seem to understand the comments on [this file](https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/servers/init.lua#L9-L12).

**Comment by williamboman** [comment, marked as answer]
Hello! Hehe nice observation. So the short answer is: backwards compatibility reasons. Long answer is: This is due to historic reasons where the installation directory used to be explicitly defined for each server, where I in some cases, for some reason, decided to use a different directory name than the name of the server. When I introduced the new `.setuo()` function , which the minimal debug configuration uses, I took the opportunity and changed it so that it will [always use the server name](https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/servers/init.lua#L196-L198) as the installation directory.

**Reply by cheeze2000** [reply]
Thank you for the clarification
